### PR TITLE
MDCT-2301 - statusing ETL script

### PR DESCRIPTION
**Pull request by lindseybaron**
## Summary
[MDCT-2301](https://qmacbis.atlassian.net/browse/MDCT-2301)

To faciliate the ability to have some entities be optional for sake of form completeness, a new object has been added to mcpar.json. This new object will need to be added to the existing formTemplates for all existing MCPAR reports

Acceptance Criteria
At the root node of each formTemplate, add the following object:

  "entities": {
    "sanctions": { "required": false },
    "accessMeasures": { "required": true },
    "qualityMeasures": { "required": true },
    "plans": { "required": true },
    "bssEntities": { "required": true }
  }
Dev Note: 
This should be done using ETL, similar to the way we did the ETL for Winter Storm updates.

### How to test

I think we'll need to test this against one of the test environments with existing data
1. in an environment with existing form templates that don't have the `entities` object, run the script
2. verify all form templates now have correct `entities` object 

### Important updates

updated [services/app-api/serverless.yml](https://github.com/Enterprise-CMCS/macpro-mdct-mcr/pull/11120/files#diff-ee790d511870504b4f2e29c38bb1f7f4e22ab59f84ea337b20159b0dfb186f39) to include new script

### Author checklist

- [ x] I have performed a self-review of my code
- [ x] I have added [thorough](https://bit.ly/3zPrxuZ) tests, if necessary
- [ ] I have updated the documentation, if necessary

### CODE REVIEW

**Review by tblackwe — APPROVED** [review]

**Review by oweingrod-coforma — CHANGES_REQUESTED** [review]

File: services/app-api/handlers/etl/formTemplateEntitiesUpdate.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        if (!formTemplateId) {
+          console.error("Could not find formTemplateId", {
+            state: metadata.state,
+            id: metadata.id,
+          });
+          continue;
+        }
+
+        // get formTemplate with formTemplateID
+        const formTemplate = await getFormTemplateFromS3(
```
[the remark this answers is not in the document]
Comment (tblackwe) [reply services/app-api/handlers/etl/formTemplateEntitiesUpdate.ts:51]: we discussed it, and this was the fastest approach. I'm not opposed to changing everything, but if it doesnt exist in dynamo it should probably be deleted, not edited.

File: services/app-api/handlers/etl/formTemplateEntitiesUpdate.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+import { DynamoDB } from "aws-sdk";
+import { Context, APIGatewayProxyResult, APIGatewayEvent } from "aws-lambda";
+import s3Lib from "../../utils/s3/s3-lib";
+import { AnyObject, S3Get, S3Put } from "../../utils/types/types";
+import { buckets } from "../../utils/constants/constants";
+
+const TABLE_NAME = process.env.MCPAR_REPORT_TABLE_NAME!;
+const BUCKET_NAME = process.env.MCPAR_FORM_BUCKET!;
+
+const ENTITIES_UPDATE_DATA = {
```
[the remark this answers is not in the document]
Comment (lindseybaron) [reply services/app-api/handlers/etl/formTemplateEntitiesUpdate.ts:11]: per our conversation, not a blocking change, just something to think about for the future.

File: services/app-api/handlers/etl/formTemplateEntitiesUpdate.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
+/* eslint-disable no-console */
+import { DynamoDB } from "aws-sdk";
+import { Context, APIGatewayProxyResult, APIGatewayEvent } from "aws-lambda";
+import s3Lib from "../../utils/s3/s3-lib";
+import { AnyObject, S3Get, S3Put } from "../../utils/types/types";
+import { buckets } from "../../utils/constants/constants";
+
+const TABLE_NAME = process.env.MCPAR_REPORT_TABLE_NAME!;
```
[the remark this answers is not in the document]
Comment (lindseybaron) [reply services/app-api/handlers/etl/formTemplateEntitiesUpdate.ts:8]: per our conversation, not necessary in this case.

File: services/app-api/handlers/etl/formTemplateEntitiesUpdate.ts [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    statusCode: 200,
+    body: JSON.stringify({
+      message: "finished ETL script",
+    }),
+  };
+};
+
+export const initialize = () => {
+  const ddbConfig: any = {};
+  const ddbEndpoint = process.env.DYNAMODB_URL;
```
[the remark this answers is not in the document]
Comment (lindseybaron) [reply services/app-api/handlers/etl/formTemplateEntitiesUpdate.ts:81]: per our conversation, probably too big of a change for this PR, but should be a TODO.

**Review by oweingrod-coforma — APPROVED** [review]

**Pull request merged**